internal/kafka/internal/services/kafka.go

```diff
@@ -164,21 +166,75 @@ func (k *kafkaService) capacityAvailableForRegionAndInstanceType(instTypeRegCapa
 	for _, kafka := range kafkas {
 		kafkaInstanceSize, e := k.kafkaConfig.GetKafkaInstanceSize(kafka.InstanceType, kafka.SizeId)
 		if e != nil {
-			return false, errors.NewWithCause(errors.ErrorGeneral, e, errMessage)
+			return false, errors.NewWithCause(errors.ErrorInstancePlanNotSupported, e, errMessage)
 		}
 		count += int64(kafkaInstanceSize.CapacityConsumed)
 	}
 
 	kafkaInstanceSize, e := k.kafkaConfig.GetKafkaInstanceSize(kafkaRequest.InstanceType, kafkaRequest.SizeId)
 	if e != nil {
-		return false, errors.NewWithCause(errors.ErrorGeneral, e, errMessage)
+		return false, errors.NewWithCause(errors.ErrorInstancePlanNotSupported, e, errMessage)
 	}
 
 	count += int64(kafkaInstanceSize.CapacityConsumed)
 
 	return instTypeRegCapacity == nil || count <= int64(*instTypeRegCapacity), nil
 }
 
+func (k *kafkaService) GetAvailableSizesInRegion(criteria *FindClusterCriteria) ([]string, *errors.ServiceError) {
+	if criteria == nil {
+		err := errors.GeneralError("unable to get available sizes in region: criteria was not specified")
+		logger.Logger.Error(err)
+		return nil, err
+	}
+
+	supportedInstanceTypes := k.kafkaConfig.SupportedInstanceTypes.Configuration
+	it, err := supportedInstanceTypes.GetKafkaInstanceTypeByID(criteria.SupportedInstanceType)
```

Comment:
to me, variable name `it` looks always like an `iterator`. Mind changing this to something more meaningful like `instanceType` or so?